### DISCUSSION TITLE
Potion effects on entities are not visible to the client

### DISCUSSION THREAD

**Discussion by VegetalDev** [Minestom/Minestom#1547] [Help]
Tested on e5d0a43.

Code used to spawn the entity is below:

```java
Entity entity = new Entity(EntityType.ZOMBIE);
entity.setInstance(player.getInstance(), player.getPosition()).join();
entity.addEffect(new Potion(PotionEffect.GLOWING, (byte)0, 10000, Potion.PARTICLES_FLAG));
```

The glowing effect does not appear on the client and no particles are visible. 

This issue seems to apply to all effect types, not just `GLOWING` (I specifically tested `INVISIBILITY` too), and appears to happen with many different values chosen for the amplifier, duration, and various flags besides the ones shown.

General info:
1) I'm testing on a fork, but it doesn't modify anything related to entity effect packets.
2) I have tested both connecting through Velocity and directly to the server, with identical results.
3) Effects work when given to players.

**Comment by mworzala** [comment]
Effects on players are handled by the client, effects on entities are handled by the server. Minestom does not do this. For example, glowing and invisibility are both entity metadata flags which need to be set (`Entity#setGlowing` and `Entity#setInvisible` respectively)

**Reply by VegetalDev** [reply]
~~`Entity#setInvisible` does not work for any entity other than armorstands; this seems to be vanilla behavior (ex. `/summon minecraft:zombie ~ ~ ~ {Invisible:1b}` does not result in an invisible zombie; `/summon minecraft:armor_stand ~ ~ ~ {Invisible:1b}` makes an invisible armor stand, as expected). Likewise, `Entity#setInvisible` on Minestom's end makes armorstands invisible.~~
Re-tested and `Entity#setInvisible` seems to work fine, not sure what the problem here was.

`/effect give @e[type=minecraft:zombie] minecraft:invisibility 1000 1` works as expected and would be visible to all players on a multiplayer server. Unless I'm missing something, the server has to be signalling to the client that the entity has invisibility. Minestom appears to do this for entities as well as players by using `EntityEffectPacket`.
